### PR TITLE
Add offline route and service worker to fix Android app install banner

### PR DESCRIPTION
**Pull request by pmusaraj**

### CODE REVIEW

**Comment by SamSaffron** [comment]
Is there a meta topic that covers the changes you made and the rationale behind it?

**Comment by gschlager** [comment]
@eviltrout Could you take a look at this when you've got some time?
https://meta.discourse.org/t/69382

**Comment by pmusaraj** [comment]
Erlend (and Falco and Gerhard above) asked me to look into this in a PM in meta. That link above from Gerhard is the thread, I'd be happy to invite the reviewer to it.

File: app/views/offline/offline.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  <title></title>
+</head>
+
+<div style="padding:100px 40px; text-align:center; font-family: Helvetica, Arial, sans-serif;">
+  <h1 style="font-size: 36px; color: #333;"><%= t 'offline.title' %></h1>
+  <p style="margin: 40px; font-size: 18px; color: #555;">
+    <%= t 'offline.offline_page_message' %>
+  </p>
+</div>
+</body>
```
Comment (tgxworld) [inline app/views/offline/offline.html.erb]: Seems like the opening `<body>` tag is missing?

File: public/service-worker.js [hunk 2]
```diff
@@ -1,4 +1,103 @@
 /*
- This service worker doesn't actually do anything!
- I'm here just to support Google Chrome App Banner on Android
+ I'm here to support Google Chrome App Banner on Android
  */
+
+/*
+ Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (tgxworld) [inline public/service-worker.js]: I'm uneasy about the license here. Why do we need it?

File: app/views/offline/offline.html.erb (hunk 1 above)
Comment (pmusaraj) [reply app/views/offline/offline.html.erb]: Yes, fixed it.

File: public/service-worker.js (hunk 2 above)
Comment (pmusaraj) [reply public/service-worker.js]: I removed it.

**Comment by SamSaffron** [comment]
OK lets try this out! merging in now thanks @pmusaraj !

**Comment by jethrogb** [comment]
I'm always getting this message after "undo close tab" in Firefox

**Pull request merged**